gtsfm/utils/prepare_nerfstudio.py

```diff
@@ -0,0 +1,162 @@
+
+    # Only supports one camera
+    camera_params = cameras[1].params
+
+    frames = []
+    for _, im_data in images.items():
+        rotation = Rot3(im_data.qvec[0], im_data.qvec[1], im_data.qvec[2], im_data.qvec[3]).matrix()
+        translation = im_data.tvec.reshape(3, 1)
+        w2c = np.concatenate([rotation, translation], 1)
+        w2c = np.concatenate([w2c, np.array([[0, 0, 0, 1]])], 0)
+        c2w = np.linalg.inv(w2c)
+
+        # Convert from COLMAP's camera coordinate system to nerfstudio's
+        c2w[0:3, 1:3] *= -1
+        c2w = c2w[np.array([1, 0, 2, 3]), :]
+        c2w[2, :] *= -1
+
+        name = Path(f"./images/{im_data.name}")
+
+        frame = {
+            "file_path": name.as_posix(),
+            "transform_matrix": c2w.tolist(),
+        }
+        frames.append(frame)
+
+    out = {
+        "fl_x": float(camera_params[0]),
+        "fl_y": float(camera_params[1]),
+        "cx": float(camera_params[2]),
+        "cy": float(camera_params[3]),
+        "w": cameras[1].width,
+        "h": cameras[1].height,
+        "camera_model": camera_model.value,
+    }
+
+    if camera_model == CameraModel.OPENCV:
+        out.update(
+            {
+                "k1": float(camera_params[4]),
+                "k2": float(camera_params[5]),
+                "p1": float(camera_params[6]),
+                "p2": float(camera_params[7]),
+            }
+        )
+    if camera_model == CameraModel.OPENCV_FISHEYE:
+        out.update(
+            {
+                "k1": float(camera_params[4]),
+                "k2": float(camera_params[5]),
+                "k3": float(camera_params[6]),
+                "k4": float(camera_params[7]),
+            }
+        )
+
+    out["frames"] = frames
+
+    with open(os.path.join(output_dir, "transforms.json"), "w", encoding="utf-8") as f:
+        json.dump(out, f, indent=4)
+
+    return cameras[1].width, cameras[1].height
+
+
+def resize_and_save_images_for_nerfstudio(images_dir, image_width, image_height, resized_images_dir):
+    """Resizes images to the size used in GTSfM and saves for nerfstudio.
+    Args:
+        images_dir: path to the original images.
+        images_width: int width of resized images
+        images_height: int height of resized images.
+        resized_images_dir: path to directory where the resized images will be saved.
+
+    """
+    image_filenames = os.listdir(images_dir)
+    for image_filename in image_filenames:
+        image_path = os.path.join(images_dir, image_filename)
+        image = io.load_image(image_path)
+        resized_image = images.resize_image(image, image_height, image_width)
+        resized_image_path = os.path.join(resized_images_dir, image_filename)
+        io.save_image(resized_image, resized_image_path)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--results_path", required=True, help="Relative or absolute path to /gtsfm/results/")
```

Comment:
nit: /gtsfm/results -> GTSFM results?

Reply:
is this the output_dir we use when running gtsfm?

Reply:
Yes